casanovo/data/db_utils.py

```diff
@@ -0,0 +1,317 @@
+"""Unique methods used within db-search mode"""
+
+import logging
+import os
+from typing import List
+
+import depthcharge.masses
+from numba import jit
+import pandas as pd
+from pyteomics import fasta, parser
+
+logger = logging.getLogger("casanovo")
+
+# CONSTANTS
+PROTON = 1.00727646677
+ISOTOPE_SPACING = 1.003355
+
+
+class ProteinDatabase:
```

Comment:
**praise:** Nice work on the refactoring. Much cleaner for the overall code flow I think.

Reply:
Thank you!